browser/ipfs/ipfs_service_factory.cc

```diff
@@ -71,7 +35,7 @@ IpfsServiceFactory* IpfsServiceFactory::GetInstance() {
 // static
 IpfsService* IpfsServiceFactory::GetForContext(
     content::BrowserContext* context) {
-  if (!IsIpfsEnabled(context))
+  if (!brave::IsRegularProfile(context) || !IsIpfsEnabled(context))
```

Comment:
curious, why this is needed. In theory, if you don't override `GetBrowserContextToUse`, default `GetBrowserContextToUse` will return `nullptr` for anything off-the-record and the service will not be instantiated. So removing this check and also removing  `GetBrowserContextToUse` should do the trick

Reply:
`GetBrowserContextToUse` override is not needed actually, removed.
`brave::IsRegularProfile(context)` checks `IsGuestSession()` also and is currently used by several of our service factories, we don't want to create instance for the regular guest profile (which is created when switching to guest happens, tho only the incognito profile of it will be used for browsing).

Reply:
I'm wondering whether `brave::IsRegularProfile` is needed at all, beacuse both Tor and Guest profiles are off-the-record. IMO we should drop it in favor of `BrowserContext::IsOffTheRecord`, wdyt?

Reply:
This way, it seems like there is no sense in using here